run-command.c

```diff
@@ -967,18 +967,24 @@ int start_command(struct child_process *cmd)
 		return -1;
 	}
 
-	if (need_in)
+	if (need_in) {
 		close(fdin[0]);
+		set_cloexec(fdin[1]);
+	}
```

Comment:
This is a very good idea. At first, though, I misread this as closing the very file descriptor in the child that the child needs to read from.

I guess it would make for another really good idea to describe a bit better what this patch is all about. In any case, the format of the commit message should also be made to conform to https://git-scm.com/docs/SubmittingPatches#describe-changes. A good example to follow is ccdc4819d56202d49e35fbfec2960e221576f862, I would think.

It also misses a [`Signed-off-by` footer](https://git-scm.com/docs/SubmittingPatches#dco), which is required by the Git project (to which we will want to contribute your fix, eventually).

Reply:
Good points, will write up in detail what the issue was with existing code and how this change fixes it. Really appreciate the example.

Reply:
I updated commit messages, hope they make things clearer.  If you have any feedback though, suggestions are welcome.